tests/Unit/NumericalAlgorithms/Interpolation/Test_MultiLinearSpanInterpolation.cpp

```diff
@@ -0,0 +1,146 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t dim, size_t num_var>
+void test() {
+  //  CAPTURE(Degree);
+
+  // pseudo random number generator
+  std::random_device rd;   // seed
+  std::mt19937 gen(rd());  // generator
+
+  std::uniform_real_distribution<> dist_func(-1., 1.);
+  std::uniform_int_distribution<size_t> dist_int(10, 20);
+
+  // Main idea: The table performs a multi-linear interpolation
+  // Each element in the table will be multi-linear function,
+  // so we can use this directly as a test.
+  // For simplicity, we will use the table to globally reproduce a
+  // single multi-linear function.
+
+
+  std::array< std::array<double, dim>, num_var> A;
+  std::array< double, num_var> b;
+
+  for (size_t i = 0; i < num_var; ++i) {
+    b[i] = dist_func(gen);
+    for (size_t j = 0; j < dim; ++j)
+      A[i][j] = dist_func(gen);
+  }
+
+  auto f = [&](std::array<double, dim> const& x) {
+    std::array<double, num_var> res{0};
```

Comment:
I'm not sure we can universally rely on `{0}` giving an entire array of `0`. I believe rules of aggregate initialization say know. We have a `make_array<Dim>(0.)` function.

We also use `0.` or `0.0` for doubles (just be consistent within each file)

Reply:
We are initializing it in the next line.